elasticapm/utils/cloud.py

```diff
@@ -47,17 +47,20 @@ def aws_metadata():
         # and will be quiet in the logs, unlike urllib3
         socket.create_connection(("169.254.169.254", 80), 0.1)
 
-        ttl_header = {"X-aws-ec2-metadata-token-ttl-seconds": "300"}
-        token_url = "http://169.254.169.254/latest/api/token"
-        token_request = http.request("PUT", token_url, headers=ttl_header, timeout=3.0)
-        token = token_request.data.decode("utf-8")
-        aws_token_header = {"X-aws-ec2-metadata-token": token} if token else {}
+        try:
+            ttl_header = {"X-aws-ec2-metadata-token-ttl-seconds": "300"}
+            token_url = "http://169.254.169.254/latest/api/token"
+            token_request = http.request("PUT", token_url, headers=ttl_header, timeout=1.0, retries=False)
+            token = token_request.data.decode("utf-8")
+            aws_token_header = {"X-aws-ec2-metadata-token": token} if token else {}
+        except Exception:
```

Comment:
Would be great if we could be a bit more specific what exceptions we want to catch here, if only for self-documenting purposes

Reply:
I think I'm going to leave this as-is. In my further explorations of IMDSv2 (and conversations with the beats team) it appears that this code is *almost* unnecessary. As in, very security-conscious users would have had to disable IMDSv1 for this code to actually be necessary.

As such, I want to limit this code's impact, whether from the ReadTimeout errors that were reported in #880, or from any decoding errors that might crop up.

I am going to add a comment about IMDSv2 so that the purpose of this block (and the fact that it's almost optional) is clear.